### PR TITLE
[DPE-5324] Increase linting rules

### PR DESCRIPTION
**Pull request by dragomirp**
Add more linting rules and change and suppress the reported issues. Added rules are:
* [flake8-bugbear](https://docs.astral.sh/ruff/rules/#flake8-bugbear-b)
* [ruff specific rules](https://docs.astral.sh/ruff/rules/#ruff-specific-rules-ruf)
* [flake8-bandit](https://docs.astral.sh/ruff/rules/#flake8-bandit-s)
* [flake8-simplify](https://docs.astral.sh/ruff/rules/#flake8-simplify-sim)
* [pyupdate](https://docs.astral.sh/ruff/rules/#pyupgrade-up)
* [flake8-type-checking](https://docs.astral.sh/ruff/rules/#flake8-type-checking-tch)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -100,13 +100,13 @@ target-version = ["py38"]
 [tool.ruff]
 # preview and explicit preview are enabled for CPY001
 preview = true
-target-version = "py38"
+target-version = "py310"
```
Comment (dragomirp) [inline pyproject.toml:103]: No libs or focal, so it is safe.

File: src/cluster.py [hunk 2]
```diff
@@ -49,6 +49,8 @@
 
 RUNNING_STATES = ["running", "streaming"]
 
+PATRONI_TIMEOUT = 10
+
```
Comment (dragomirp) [inline src/cluster.py:53]: Bandit rules want timeouts for HTTP calls.

File: src/cluster_topology_observer.py [hunk 3]
```diff
@@ -85,13 +86,14 @@ def start_observer(self):
                 self._charm.unit.name,
                 self._charm.charm_dir,
             ],
-            stdout=open(LOG_FILE_PATH, "a"),
+            # File shouldn't close
+            stdout=open(LOG_FILE_PATH, "a"),  # noqa: SIM115
```
Comment (dragomirp) [inline src/cluster_topology_observer.py:90]: We should most probably set the output in the running instance instead or switch to a proper service.

**Review by marceloneppel — APPROVED** [review]
LGTM! I liked the HTTP calls timeout enforcement.

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**